src/pandasaurus/utils/sparql_queries.py

```diff
@@ -1,34 +1,50 @@
 from typing import List
 
 
-def get_simple_enrichment_query(seed_list: List[str]) -> str:
-    # TODO Add missing implementation
-    pass
+# TODO might need to update the enrichment query method names
+def get_simple_enrichment_query(
+    s_iri_list: List[str], o_iri_list: List[str], property_list: List[str]
+) -> str:
+    return (
+        f"SELECT ?s ?s_label ?p ?o ?o_label WHERE {{ GRAPH <http://reasoner.renci.org/redundant> {{"
+        f"VALUES ?s {{ {' '.join(s_iri_list)} }} VALUES ?o {{ {' '.join(o_iri_list)} }} "
+        f"VALUES ?p {{ {' '.join(property_list)} }}"
+        f"?s ?p ?o. }} ?s rdfs:label ?s_label. ?o rdfs:label ?o_label. FILTER(?s != ?o)}} LIMIT 1000"
+    )
 
 
 def get_minimal_enrichment_query(seed_list: List[str]) -> str:
     # TODO Add missing implementation
     pass
 
 
-def get_full_enrichment_query(seed_list: List[str]) -> str:
-    # TODO Add missing implementation
-    pass
+def get_full_enrichment_query(s_iri_list: List[str], o_iri_list: List[str]) -> str:
+    return (
+        f"SELECT ?s ?s_label ?p ?o ?o_label WHERE {{ GRAPH <http://reasoner.renci.org/nonredundant> {{"
+        f"VALUES ?s {{ {' '.join(s_iri_list)} }} VALUES ?o {{ {' '.join(o_iri_list)} }} "
+        f"?s rdfs:subClassOf* ?o. }} ?s rdfs:label ?s_label. ?o rdfs:label ?o_label FILTER(?s != ?o)}} LIMIT 1000"
+    )
 
 
-def get_contextual_enrichment_query(seed_list: List[str]) -> str:
-    # TODO Add missing implementation
-    pass
+def get_contextual_enrichment_query(context_list: List[str]) -> str:
+    # BFO:0000050 , part of
+    return (
+        f"SELECT * WHERE {{ VALUES ?context {{{' '.join(context_list)} }} ?term BFO:0000050 ?context. "
+        f"?term rdfs:label ?label }} LIMIT 1000"
+    )
 
 
 def get_curie_prefix_validation_query(seed_list: List[str]) -> str:
     # TODO Add missing implementation
     pass
 
 
-def get_curie_validation_query(seed_list: List[str]) -> str:
-    # TODO Add missing implementation
-    pass
+def get_label_query(term_iri_list: List[str]) -> str:
+    """Used for CURIE validation"""
+    return (
+        f"SELECT ?term ?label where {{ ?term rdf:type ?type. OPTIONAL {{?term rdfs:label ?label.}} "
```

Comment:
Why do you need the part `?term rdf:type ?type`? You can specify the type as `owl:Class`.

Reply:
I didn't think it would make a difference in terms of performance, since we are providing a term list to the query anyway. But I guess it would be better to use `owl:Class` if a term would have multiple type.

Reply:
Every triple you add is part of the query, and so to the complexity calculation. It's better to add more specific triples than only using variables if unnecessary.